Upload dataset cs vfile

robertmunne opened the pull request.

robertmunne committed:
new field to add csv created

robertmunne committed:
separador pel csv

robertmunne committed:
dataset content (csv) added

robertmunne committed:
elimino variables innecessàries.

rogargon commented on src/app/dataset/dataset-create/dataset-create.component.html:
```diff
@@ -18,11 +18,28 @@ <h4 class="display-4">Create a dataset</h4>
+    </div>
+
+    <!-- Dataset separador -->
+    <div class="form-group">
```
It is better to not let users define this value because it depends on the file that has been uploaded. You can get the content type of the file using `File.type`, more details:
https://developer.mozilla.org/es/docs/Web/API/File/type

rogargon commented on src/app/dataset/dataset-create/dataset-create.component.html:
```diff
@@ -18,11 +18,28 @@ <h4 class="display-4">Create a dataset</h4>
+    <!-- Dataset separador -->
+    <div class="form-group">
+      <label class="control-label" for="contentType">Separador</label>
+      <input id="contentType" name="contentType" type="text" class="form-control"
```
It is better to not let users define this value because it depends on the file that has been uploaded. You can get the content type of the file using File.type, more details:
https://developer.mozilla.org/es/docs/Web/API/File/type

robertmunne commented:
Hola,

He mirat el Type de l'exemple amb JS, entenc que vols que flitri en el
component ts, algo així:

addDataFile(event): void {
  const fileList: FileList = event.target.files;
  const FileToUpload: File = fileList[0];

  if (FileToUpload.type === 'text/plain' ){
    const reader = new FileReader();
    reader.readAsText(FileToUpload);
    reader.onloadend = (e) => {
      this.dataset.content = reader.result;
    };
  } else {
    throwError('errorMessage');
  }

}.

Gràcies


El vie, 11 dic 2020 a las 11:20, Roberto García (<notifications@github.com>)
escribió:

> *@rogargon* commented on this pull request.
> ------------------------------
>
> In src/app/dataset/dataset-create/dataset-create.component.html
> <https://github.com/UdL-EPS-SoftArch/AgridataHub/pull/28#discussion_r540840943>
> :
>
> > @@ -18,11 +18,28 @@ <h4 class="display-4">Create a dataset</h4>
>               [(ngModel)]="dataset.description">
>      </div>
>
> +    <!-- Dataset content -->
> +    <div class="form-group">
> +      <label class="control-label" for="inputFile">CSV</label>
> +      <div class="col-md-9">
> +        <input type="file" id="inputFile" autofocus="" (change)="addDataFile($event)" #file/>
> +      </div>
> +
> +    </div>
> +
> +    <!-- Dataset separador -->
> +    <div class="form-group">
> +      <label class="control-label" for="contentType">Separador</label>
> +      <input id="contentType" name="contentType" type="text" class="form-control"
>
> It is better to not let users define this value because it depends on the
> file that has been uploaded. You can get the content type of the file using
> File.type, more details:
> https://developer.mozilla.org/es/docs/Web/API/File/type
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UdL-EPS-SoftArch/AgridataHub/pull/28#pullrequestreview-549970459>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AODONV5DQ27XCAIZLWOZDWLSUHW5NANCNFSM4UWITRNA>
> .
>

robertmunne committed:
Pujo a la meva branca les proves que he fet amb js que no funcionen i el codi amb ts que penso que és el que es demana.

rogargon commented:
No need to filter files for upload, just fill dataset.contentType from the fileToUpload.type

robertmunne committed:
Fets els canvis de la última issue.
Com a observació vaig demanar en pantalla el contentType per l'escenari semblava que demanava el separador de camp.

Adjunto escenari: Scenario: Provider upload your datasets
                      Given I login as "provider" with password "password"
                      When I upload a dataset with title "test.csv" and description "description" and contentType ","
                      Then The dataset contains a file with title "test.csv"
                      And The dataset content contains "test.csv" content
                      And The dataset contentType is ","
